Dockerfile

```
@@ -0,0 +1,45 @@
+
+# Create non-root user that mirrors the host developer UID/GID
+RUN groupadd --gid "${GID}" "${USERNAME}" \
+    && useradd --uid "${UID}" --gid "${GID}" --create-home "${USERNAME}"
+
+# Create a dedicated virtual environment for Python packages
+RUN python -m venv /opt/pulumi-venv
+ENV VIRTUAL_ENV=/opt/pulumi-venv
+ENV PATH="$VIRTUAL_ENV/bin:/home/${USERNAME}/.pulumi/bin:${PATH}"
+
+# Install Pulumi CLI
+RUN curl -fsSL https://get.pulumi.com | sh
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)